src/transformers/modeling_tf_utils.py

```diff
@@ -80,17 +80,18 @@
 
 logger = logging.get_logger(__name__)
 
-if "TF_USE_LEGACY_KERAS" not in os.environ:
-    os.environ["TF_USE_LEGACY_KERAS"] = "1"  # Compatibility fix to make sure tf.keras stays at Keras 2
-elif os.environ["TF_USE_LEGACY_KERAS"] != "1":
-    logger.warning(
-        "Transformers is only compatible with Keras 2, but you have explicitly set `TF_USE_LEGACY_KERAS` to `0`. "
-        "This may result in unexpected behaviour or errors if Keras 3 objects are passed to Transformers models."
-    )
-
 try:
     import tf_keras as keras
     from tf_keras import backend as K
+
+    if "TF_USE_LEGACY_KERAS" not in os.environ:
+        os.environ["TF_USE_LEGACY_KERAS"] = "1"  # Compatibility fix to make sure tf.keras stays at Keras 2
+    elif os.environ.get("TF_USE_LEGACY_KERAS", None) not in ("true", "True", "1"):
```

Comment:
notice this was changes from just checking `1` to match what tensorflow checks for

https://github.com/tensorflow/tensorflow/blob/de7cbafcf424869fbf1d7aa9f9b39c87e23329a6/tensorflow/python/util/lazy_loader.py#L129